File: LexicalAnalyzer.c
#include<stdio.h>
#include<ctype.h>
#include<string.h>

char keywords[10][10] = {
    "int","float","char","if","else",
    "while","for","return","void","printf"
};

int isKeyword(char str[]) {
    for(int i=0;i<10;i++) {
        if(strcmp(str, keywords[i])==0)
            return 1;
    }
    return 0;
}

int main() {
    char str[100];
    printf("Enter a line of code:\n");
    gets(str);

    int i=0;
    while(str[i] != '\0') {

        // Identifier or keyword
        if(isalpha(str[i])) {
            char temp[20];
            int j=0;

            while(isalnum(str[i])) {
                temp[j++] = str[i++];
            }
            temp[j]='\0';

            if(isKeyword(temp))
                printf("%s is Keyword\n", temp);
            else
                printf("%s is Identifier\n", temp);
        }

        // Number
        else if(isdigit(str[i])) {
            char temp[20];
            int j=0;

            while(isdigit(str[i])) {
                temp[j++] = str[i++];
            }
            temp[j]='\0';

            printf("%s is Constant\n", temp);
        }

        // Operator
        else if(strchr("+-*/=%", str[i])) {
            printf("%c is Operator\n", str[i]);
            i++;
        }

        // Special symbols
        else if(strchr("();{}", str[i])) {
            printf("%c is Special Symbol\n", str[i]);
            i++;
        }

        else {
            i++;
        }
    }

    return 0;
}

File: LexicalAnalyzer.java
import java.util.*;

public class LexicalAnalyzer {

    static String keywords[] = {
        "int","float","char","if","else",
        "while","for","return","void","printf"
    };

    static boolean isKeyword(String str) {
        for(String k : keywords) {
            if(k.equals(str))
                return true;
        }
        return false;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter a line of code:");
        String line = sc.nextLine();

        int i = 0;
        while(i < line.length()) {

            // Identifier or keyword
            if(Character.isLetter(line.charAt(i))) {
                String temp = "";

                while(i < line.length() && 
                      Character.isLetterOrDigit(line.charAt(i))) {
                    temp += line.charAt(i);
                    i++;
                }

                if(isKeyword(temp))
                    System.out.println(temp + " is Keyword");
                else
                    System.out.println(temp + " is Identifier");
            }

            // Number
            else if(Character.isDigit(line.charAt(i))) {
                String temp = "";

                while(i < line.length() && 
                      Character.isDigit(line.charAt(i))) {
                    temp += line.charAt(i);
                    i++;
                }

                System.out.println(temp + " is Constant");
            }

            // Operator
            else if("+-*/=%".indexOf(line.charAt(i)) != -1) {
                System.out.println(line.charAt(i) + " is Operator");
                i++;
            }

            // Special symbols
            else if("();{}".indexOf(line.charAt(i)) != -1) {
                System.out.println(line.charAt(i) + " is Special Symbol");
                i++;
            }

            else {
                i++;
            }
        }

        sc.close();
    }
}